sign/eddsa/eddsa_test.go

```diff
@@ -130,6 +130,28 @@ func TestEdDSASigningMalleability(t *testing.T) {
 	assert.Error(t, Verify(ed.Public, msg, sig))
 }
 
+// Test non-canonical keys
+func TestEdDSASigningNonCanonical(t *testing.T) {
+	var nonCanonicalPk []byte = []byte{0xf6, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+		0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+		0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x7f}
+	suite := edwards25519.NewBlakeSHA256Ed25519()
+	ed := NewEdDSA(suite.RandomStream())
```

Comment:
```suggestion
	ed := NewEdDSA(random.New())
```

Then you don't need the `suite`...

Reply:
I wrote that because that what is used on TestEdDSASigningRandom, thus everything remains uniform.

Reply:
OK - then I think we need to change the `TestEdDSASigningRandom` ;)

Reply:
Don't worry, on this PR I have change them all.